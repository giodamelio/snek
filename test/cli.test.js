const { spawnSync } = require('child_process');

function runSnek(...args) {
  const command = spawnSync('node', ['./bin/snek.js', ...args]);
  command.stdout = command.stdout && command.stdout.toString();
  command.stderr = command.stderr && command.stderr.toString();
  return command;
}

it('print help', () => {
  const { status, stdout, stderr } = runSnek('-h');

  expect(status).toBe(0);
  expect(stdout).toMatchSnapshot();
  expect(stderr).toBe('');
});
