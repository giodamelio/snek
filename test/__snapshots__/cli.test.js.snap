// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`print default snek 1`] = `
"csegment [ '‚†Å', '‚†É', '‚†á', '‚†ß' ]
move right
segment [ '‚†Ñ', '‚†§', '‚†¥', '‚†º' ]
move right
segment [ '‚†Å', '‚†â' ]
move right
[0m"
`;

exports[`print help 1`] = `
"snek version 0.2.0
usage: snek
       snek tiny
       snek tall

       You can also add modifiers to your sneks to make them do things
       --fast  Doubles the sneks speed
[0m[?1000l[?1002l[?1003l[?1004l"
`;

exports[`print tall snek 1`] = `
"csegment [ '‚†Å', '‚†É', '‚†á', '‚°á' ]
move down
segment [ '‚†Å', '‚†É', '‚†á', '‚°á', '‚£á' ]
move right
segment [ '‚¢Ä', '‚£Ä' ]
move right
segment [ '‚°Ä', '‚¢†', '‚¢∞', '‚¢∏' ]
move up
segment [ '‚°Ä', '‚¢†', '‚¢∞', '‚¢∏', '‚°è' ]
move right
segment [ '‚†Å', '‚†â' ]
move right
[0m"
`;

exports[`print tiny snek 1`] = `
"csegment [ '‚†Å', '‚†É', '‚†á', '‚†ß' ]
move right
segment [ '‚†Ñ', '‚†§', '‚†¥', '‚†º' ]
move right
segment [ '‚†Å', '‚†â' ]
move right
[0m"
`;
